isis/tests/FunctionalTestsHicrop.cpp

```diff
@@ -0,0 +1,438 @@
+  EXPECT_EQ((int)results.findKeyword("NumberOfLinesCropped"), 54);
+  EXPECT_EQ((int)results.findKeyword("OututStartingLine"), 55);
+  EXPECT_EQ((int)results.findKeyword("OututEndingLine"), 1056);  
+  EXPECT_EQ((int)results.findKeyword("OututLineCount"), 1002); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartTime"), "2008-05-17T09:37:24.7300819");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopTime"), "2008-05-17T09:37:31.0666673");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartClock"), "4/0895484265.14186"); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopClock"), "4/0895484271.36245"); 
+ 
+  Cube oCube(outCubeFileName); 
+  PvlGroup oCubeInstrument = testCube->label()->findObject("IsisCube").findGroup("Instrument");
+  
+  EXPECT_EQ(oCube.lineCount(), 1002);
+  EXPECT_EQ(oCube.sampleCount(), 1204);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("StartTime"), "2008-05-17T09:37:24.7300819");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("StopTime"), "2008-05-17T09:37:31.0666673");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("SpacecraftClockStartCount"), "895484264:57342");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)oCubeInstrument.findKeyword("SpacecraftClockStopCount"), "895484272:12777");
+  std::cout << "DONE" << std::endl;
+}
+
+
+TEST_F(MroCube, FunctionalTestHicropCropByLine) {
+  setInstrument("-74999", "HIRISE", "MARS RECONNAISSANCE ORBITER");
+  
+  QTemporaryDir prefix;
+  QString outCubeFileName = prefix.path() + "/outTemp.cub";
+  
+  QVector<QString> args = {"to="+outCubeFileName, "source=LINE", "LINE=55", "NLINES=1002", "CK="+ckPath,
+  "SCLK="+sclkPath, "LSK="+lskPath};
+  
+  UserInterface options(APP_XML, args);
+  Pvl *logs = new Pvl();
+  try {
+     hicrop(testCube, options, logs);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to open image: " << e.what() << std::endl;
+  }
+
+  PvlGroup results = logs->findGroup("Results");
+  
+  EXPECT_EQ((int)results.findKeyword("InputLines"), 1056);
+  EXPECT_EQ((int)results.findKeyword("NumberOfLinesCropped"), 54);
+  EXPECT_EQ((int)results.findKeyword("OututStartingLine"), 55);
+  EXPECT_EQ((int)results.findKeyword("OututEndingLine"), 1056);  
+  EXPECT_EQ((int)results.findKeyword("OututLineCount"), 1002); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartTime"), "2008-05-17T09:37:24.7892562");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopTime"), "2008-05-17T09:37:24.8845088");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStartClock"), "4/0895484264.57678"); 
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, (QString)results.findKeyword("OututStopClock"), "4/0895484264.63921"); 
```

Comment:
Note the typo: `Outut`. This is part of the app, decided not to fix it as it may break people's scripts.